src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1710,14 +1697,12 @@ void enableSSL(String host,
                 tmf.init(ks);
                 tm = tmf.getTrustManagers();
 
-                // if the host name in cert provided use it or use the host name Only if it is not FIPS
-                if (!isFips) {
-                    if (null != hostNameInCertificate) {
-                        tm = new TrustManager[] {new HostNameOverrideX509TrustManager(this, (X509TrustManager) tm[0], hostNameInCertificate)};
-                    }
-                    else {
-                        tm = new TrustManager[] {new HostNameOverrideX509TrustManager(this, (X509TrustManager) tm[0], host)};
-                    }
+                // if the host name in cert provided use it or use the host name 
```

Comment:
In case of FIPS our driver should delegate responsibility of validating certificate to JVM. 

In this case we are using our own TrustManager i.e. `HostNameOverrideX509TrustManager` which will throw following exception : 

````
FIPS mode: only SunJSSE TrustManagers may be used
````
Ref: [OpenJDK: SSLContextImpl](https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/sun/security/ssl/SSLContextImpl.java#L126)

If I am correctly remember in order to remove error `FIPS mode: only SunJSSE TrustManagers may be used` we review our FIPS implementation and added fips attribute. 

If you want to get rid of fips attribute then you have to come up with function which will detect if JVM is in FIPS mode or not. You can find logic in our test cases as well as on internet. 

I will suggest instead of merging back in master / dev branch you can come up with `FIPS` or `FIPSRnD` branch and test with different FIPS implementation. If users are comfortable with then you can merge back in main stream

Remember to test with at-least following FIPS implementations: 

1. BouncyCastle 
1. IBM
1. OpenJDK 9  (Not Tested yet)